ArduPlane/mode_takeoff.cpp

```diff
@@ -105,20 +105,21 @@ void ModeTakeoff::update()
         }
     }
 
-    // we finish the initial level takeoff if we climb past
-    // TKOFF_LVL_ALT or we pass the target location. The check for
-    // target location prevents us flying forever if we can't climb
-    if (plane.flight_stage == AP_FixedWing::FlightStage::TAKEOFF &&
-        (plane.current_loc.alt - start_loc.alt >= level_alt*100 ||
-         start_loc.get_distance(plane.current_loc) >= target_dist)) {
-        // reached level alt, re-calculate bearing to cope with systems with no compass
-        // or with poor initial compass
-        float direction = start_loc.get_bearing_to(plane.current_loc) * 0.01;
-        float dist_done = start_loc.get_distance(plane.current_loc);
-
-        plane.next_WP_loc = plane.current_loc;
-        plane.next_WP_loc.offset_bearing(direction, MAX(dist-dist_done, 0));
-        plane.next_WP_loc.alt = start_loc.alt + target_alt*100.0;
+        // we finish the initial level takeoff if we climb past
+        // TKOFF_LVL_ALT or we pass the target location. The check for
+        // target location prevents us flying forever if we can't climb
+        // reset the loiter waypoint target to be correct bearing and dist 
+        // from starting location in case original yaw used to set it was off due to EKF 
+        // reset or compass interference from max throttle
+        if (plane.flight_stage == AP_FixedWing::FlightStage::TAKEOFF &&
+            (plane.current_loc.alt - start_loc.alt >= level_alt*100 ||
+             start_loc.get_distance(plane.current_loc) >= dist)) {
+            // reset the target loiter waypoint using current yaw which should be close to correct starting heading
+            const float direction = degrees(ahrs.yaw);
+            const float distance = dist- start_loc.get_distance(plane.current_loc);
```

Comment:
`dist - start_loc.get_distance(plane.current_loc);` missing a space after `dist`, maybe the code style commit fixes.

Reply:
changed direction

Reply:
changed